zaza/openstack/charm_tests/ironic/setup.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Code for configuring ironic."""
+
+import copy
+import os
+
+import zaza.openstack.charm_tests.glance.setup as glance_setup
+import zaza.openstack.utilities.openstack as openstack_utils
+from zaza.openstack.utilities import (
+    cli as cli_utils,
+)
+import zaza.model as zaza_model
+
+
+FLAVORS = {
+    'bm1.small': {
+        'flavorid': 2,
+        'ram': 2048,
+        'disk': 20,
+        'vcpus': 1,
+        'properties': {
+            "resources:CUSTOM_BAREMETAL1_SMALL": 1,
+        },
+    },
+    'bm1.medium': {
+        'flavorid': 3,
+        'ram': 4096,
+        'disk': 40,
+        'vcpus': 2,
+        'properties': {
+            "resources:CUSTOM_BAREMETAL1_MEDIUM": 1,
+        },
+    },
+    'bm1.large': {
+        'flavorid': 4,
+        'ram': 8192,
+        'disk': 40,
+        'vcpus': 4,
+        'properties': {
+            "resources:CUSTOM_BAREMETAL1_LARGE": 1,
+        },
+    },
+    'bm1.tempest': {
+        'flavorid': 6,
+        'ram': 256,
+        'disk': 1,
+        'vcpus': 1,
+        'properties': {
+            "resources:CUSTOM_BAREMETAL1_TEMPEST": 1,
+        },
+    },
+    'bm2.tempest': {
+        'flavorid': 7,
+        'ram': 512,
+        'disk': 1,
+        'vcpus': 1,
+        'properties': {
+            "resources:CUSTOM_BAREMETAL2_TEMPEST": 1,
+        },
+    },
+}
+
+
+def add_ironic_deployment_image(initrd_url=None, kernel_url=None):
+    """Add Ironic deploy images to glance.
+
+    :param initrd_url: URL where the ari image resides
+    :type initrd_url: str
+    :param kernel_url: URL where the aki image resides
+    :type kernel_url: str
+    """
+    base_name = 'ironic-deploy'
+    initrd_name = "{}-initrd".format(base_name)
+    vmlinuz_name = "{}-vmlinuz".format(base_name)
+    if not initrd_url:
+        initrd_url = os.environ.get('FUNCTEST_IRONIC_DEPLOY_INITRD', None)
```

Comment:
Please make all variables begin with `TEST_`. We have been standardizing on variable names.